retriever/lib/compile.py

```diff
@@ -1,199 +1,59 @@
+def compile_json(json_file, debug=False):
     """
     Function to compile JSON script files to python scripts
     The scripts are created with `retriever new_json <script_name>` using
     command line
     """
-    json_object = {}
-    source_encoding = "latin-1"
+    json_object = OrderedDict()
+    json_file = str(json_file) + ".json"
+
     try:
-        json_object = json.load(open(json_file + ".json", "r"))
+        json_object = json.load(open(json_file, "r"))
     except ValueError:
         pass
-    if type(json_object) is not dict:
-        return
-    if "retriever" not in json_object.keys():
-        # Compile only files that have retriever key
-        return
-
-    values = {}
-    values['urls'] = {}
-
-    keys_to_ignore = ["template"]
-
-    for (key, value) in json_object.items():
-
-        if key == "title":
-            values["title"] = "\"" + str(value) + "\""
-
-        elif key == "name":
-            values["name"] = "\"" + str(value) + "\""
-
-        elif key == "description":
-            values["description"] = "\"" + str(value) + "\""
-
-        elif key == "addendum":
-            values["addendum"] = "\"" + str(value) + "\""
-
-        elif key == "homepage":
-            values["ref"] = "\"" + str(value) + "\""
-
-        elif key == "citation":
-            values["citation"] = "\"" + str(value) + "\""
-
-        elif key == "licenses":
-            values["licenses"] = value
-
-        elif key == "keywords":
-            values["keywords"] = value
-
-        elif key == "version":
-            values["version"] = "\"" + str(value) + "\""
-
-        elif key == "encoding":
-            values["encoding"] = "\"" + str(value) + "\""
-            # Adding the key 'encoding'
-            source_encoding = str(value)
-
-        elif key == "retriever_minimum_version":
-            values["retriever_minimum_version"] = "\"" + str(value) + "\""
-
-        elif key == "message":
-            values["message"] = "\"" + str(value) + "\""
-
-        elif key == "resources":
-            # Array of table objects
-            tables = {}
-            for table in value:
-                # Maintain a dict for table keys and values
-                table_dict = {}
-
-                try:
-                    values['urls'][table['name']] = table['url']
-                except Exception as e:
-                    print(e, "\nError in reading table: " + table)
-                    continue
-
-                if table["schema"] == {} and table["dialect"] == {}:
-                    continue
-
-                for (t_key, t_val) in table.items():
-
-                    if t_key == "dialect":
-                        add_dialect(table_dict, table)
-
-                    elif t_key == "schema":
-                        add_schema(table_dict, table)
-
-                tables[table["name"]] = table_dict
+    if type(json_object) is dict and "resources" in json_object.keys():
+        if "format" not in json_object:
```

Comment:
based on the corrections we made on the Spatialtables branch, this shouldn't be here